### PR TITLE
Three state button

### PR DESCRIPTION
**Pull request by EkaterinaPoddubskaya**
Created three-state button via webix.protoUI
Enabled sorting/unsorting of film list by clicking that button

### CODE REVIEW

File: scripts/main.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+let button_state_index = 0;
+
+webix.protoUI({
+    name:"threeStateButton",
+    $init(config) {
+        this.$ready.push(() => {
+            this.define("label", config.states[config.state]);
+            webix.html.addCss(this.getNode(), BUTTON_STATE[button_state_index].style);
+        });
+        this.attachEvent("onItemClick", () => {
```
Comment (VeraOrekhXB) [inline scripts/main.js:16]: The event has to be detached on `onDestruct` event

File: scripts/main.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+            webix.html.removeCss(this.getNode(), BUTTON_STATE[button_state_index].style);
```
Comment (VeraOrekhXB) [inline scripts/main.js:17]: You may add a method, for example, `changeState`
It makes it easier & quicker to understand what is going on + it's reusable and could be called on another even handler too

File: scripts/main.js [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        [BUTTON_STATE[0].id]: "Off", 
+        [BUTTON_STATE[1].id]: "SortAsc", 
+        [BUTTON_STATE[2].id]: "Sort Desc" 
+    }, 
+    state: BUTTON_STATE[0].id, 
+    on:{
+        onStateChange(state) {
+            switch(state) {
+                case BUTTON_STATE[1].id:
+                    $$("filmList").sort("#title#", "asc");
```
Comment (VeraOrekhXB) [inline scripts/main.js:42]: It's better to put `$$("filmList")` value into variable

File: scripts/main.js [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    name:"threeStateButton",
+    $init(config) {
+        this.$ready.push(() => {
+            this.define("label", config.states[config.state]);
+            webix.html.addCss(this.getNode(), BUTTON_STATE[button_state_index].style);
+        });
+        this.attachEvent("onItemClick", () => {
+            webix.html.removeCss(this.getNode(), BUTTON_STATE[button_state_index].style);
+            button_state_index = (button_state_index + 1) % BUTTON_STATE.length;
+            webix.html.addCss(this.getNode(), BUTTON_STATE[button_state_index].style);
```
Comment (VeraOrekhXB) [inline scripts/main.js:19]: `this.getNode()` put into variable

**Review by VeraOrekhXB — CHANGES_REQUESTED** [review]

**Comment by EkaterinaPoddubskaya** [comment]
I din't put config.state into a variable in changeState function because it's value changes throughout the function and I'll have to reasign it twice

File: scripts/threeStateButton.js [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+        if (invalidMessage) throw new Error(invalidMessage);
+        else return value;
+    },
+    state_setter: (value) => {
+        if (typeof value === 'number' && value >= 0 && value < 3) return value;
+        else throw new Error("State is not a number type");
+    },
+    $init(config) {
+        this.$ready.push(() => {
+            config.state = this.config.state;
```
Comment (VeraOrekhXB) [inline scripts/threeStateButton.js:48]: What is the purpose of the string?

**Review by VeraOrekhXB — CHANGES_REQUESTED** [review]

**Review by VeraOrekhXB — APPROVED** [review]

**Pull request merged**